packages/astro/src/compiler/codegen/utils.ts

```diff
@@ -18,3 +21,21 @@ export function isImportMetaDeclaration(declaration: VariableDeclarator, metaNam
   if (metaName && (init.callee.property.type !== 'Identifier' || init.callee.property.name !== metaName)) return false;
   return true;
 }
+
+const warnableRelativeValues = new Set([
+  'img+src',
+  'a+href',
+  'script+src',
+  'link+href',
+  'source+srcset'
+]);
+
+const nonRelative = /^(?!(https?|\/))/;
```

Comment:
The variable name here is `nonRelative`, but it appears designed to match _relative_  URLs. Is this correct? Should it be renamed?

Also, according to [RFC 3987 ](http://www.faqs.org/rfcs/rfc3987.html), a valid scheme matches `ALPHA *( ALPHA / DIGIT / "+" / "-" / "." )`

Therefore, a check for non-relative URLs might be:

```suggestion
const nonRelative = /^([A-Za-z][+-.0-9A-Za-z]*:|\/)/;
```

_☝️ I ordered the matches by character code. That’s why `+` comes before `0-9`, and `0-9` comes before `A-Z`._

Reply:
This matches relative links 🤦‍♂️, I'll rename and take your suggestion